java/config/checkstyle/checkstyle.xml

```diff
@@ -26,56 +26,64 @@
   <property name="charset" value="UTF-8"/>
-
+  <property name="tabWidth" value="4"/>
+  <property name="fileExtensions" value="java"/>
   <property name="severity" value="${org.checkstyle.google.severity}" default="warning"/>
 
-  <!-- It's standard practice to separate any exceptions/suppressions into a separate XML file. -->
-  <module name="SuppressionFilter">
-    <property name="file" value="${config_loc}/suppressions.xml" />
-  </module>
-
-  <property name="fileExtensions" value="java, properties, xml"/>
-  <!-- Excludes all 'module-info.java' files              -->
+  <!-- Fine-grained exclusion rules -->
   <!-- See https://checkstyle.org/filefilters/index.html -->
   <module name="BeforeExecutionExclusionFileFilter">
+    <!-- Exclude all 'module-info.java' files -->
     <property name="fileNamePattern" value="module\-info\.java$"/>
   </module>
 
-  <module name="SuppressWarningsFilter"/>
+  <!-- Warning Suppression -->
 
-  <!-- https://checkstyle.org/filters/suppressionfilter.html -->
+  <!-- It's standard practice to place suppression rules in a separate XML file. -->
+  <!-- See https://checkstyle.org/filters/suppressionfilter.html -->
   <module name="SuppressionFilter">
-    <property name="file" value="${org.checkstyle.google.suppressionfilter.config}"
-           default="checkstyle-suppressions.xml" />
+    <property name="file" value="${config_loc}/suppressions.xml" />
     <property name="optional" value="true"/>
   </module>
 
-  <module name="SuppressWarningsFilter"/>
-
-  <!-- https://checkstyle.org/filters/suppresswithnearbytextfilter.html -->
+  <!-- Allows you to suppress checkstyle warnings on a single line by writing '// CHECKSTYLE.SUPPRESS: <RULE> '-->
+  <!-- See https://checkstyle.org/filters/suppresswithnearbytextfilter.html -->
   <module name="SuppressWithNearbyTextFilter">
-    <property name="nearbyTextPattern"
-              value="CHECKSTYLE.SUPPRESS\: (\w+) for ([+-]\d+) lines"/>
+    <property name="nearbyTextPattern" value="CHECKSTYLE.SUPPRESS\: (\w+)"/>
     <property name="checkPattern" value="$1"/>
-    <property name="lineRange" value="$2"/>
   </module>
 
-  <!-- Checks for whitespace                               -->
-  <!-- See http://checkstyle.org/checks/whitespace/index.html -->
+  <!-- Basic Formatting Checks -->
+
+  <!-- Reject tabs in source code -->
+  <!-- See https://checkstyle.org/checks/whitespace/filetabcharacter.html -->
   <module name="FileTabCharacter">
     <property name="eachLine" value="true"/>
   </module>
 
+  <!-- Enforce a line length limit of 120 columns in all '.java' files. -->
+  <!-- See https://checkstyle.org/checks/sizes/linelength.html -->
   <module name="LineLength">
     <property name="fileExtensions" value="java"/>
     <property name="max" value="120"/>
-    <property name="ignorePattern"
-             value="^package.*|^import.*|^\s*&quot;|href\s*=\s*&quot;[^&quot;]*&quot;|http://|https://|ftp://"/>
```

Comment:
Simplified this regex, and removed the rule allowing `href` to go beyond 120 columns.
If we ever need that for some reason, we can revisit.